homeassistant/components/shelly/sensor.py

```diff
@@ -785,3 +828,8 @@ def native_value(self) -> StateType:
     def native_unit_of_measurement(self) -> str | None:
         """Return the unit of measurement of the sensor, if any."""
         return self.entity_description.native_unit_of_measurement
+
+    @property
+    def suggested_display_precision(self) -> int | None:
+        """Return the precision of the sensor."""
+        return self.entity_description.suggested_display_precision
```

Comment:
```suggestion
```

We should be able to use the parent's version of this instead

https://github.com/home-assistant/core/blob/1759f58fc110b6c3ad81d859d6e5716182ad29d2/homeassistant/components/sensor/__init__.py#L372